spyder/utils/ipython/spyder_kernel.py

```diff
@@ -110,7 +110,7 @@ def get_namespace_view(self):
         settings = self.namespace_view_settings
         if settings:
             ns = self._get_current_namespace()
-            view = make_remote_view(ns, settings, EXCLUDED_NAMES)
+            view = repr(make_remote_view(ns, settings, EXCLUDED_NAMES))
             return view
```

Comment:
Python makes this function to return `None` when there are no settings (which happens when there's no Variable Explorer in the window). That's why `test_browse_history_dbg` is failing.

That was working before (implicitly) because we were returning Python objects (in this case a dict). But it fails now because we are returning strings (through `repr`).

So we need to add here

    else:
        repr(None)